robocin/geometry/README.md

```diff
@@ -0,0 +1,185 @@
+# geometry
+
+A collection of geometric classes.
+
+## Table of Contents
+
+- [Point2D](#point2d)
+
+<a name="point2d"></a>
+
+## [`Point2D`](point2d.h)
+
+The `Point2D` templated struct represents a 2-dimensional point (vector) in a Cartesian coordinate system. It provides
+various methods and operators for manipulating and performing calculations with 2d-points.
+
+### Member Types
+
+- `value_type`: The type of the point's coordinates.
+- `reference`: A reference to `value_type`.
+- `size_type`: An unsigned integer type used for size and indexing.
+- `iterator`: An iterator type for iterating over mutable `Point2D` objects.
+- `const_iterator`: An iterator type for iterating over const `Point2D` objects.
+- `reverse_iterator`: A reverse iterator type for reverse iteration over mutable `Point2D` objects.
+- `const_reverse_iterator`: A reverse iterator type for reverse iteration over const `Point2D` objects.
+
+### Constructors
+
+- `Point2D()`: Default constructor. Initializes the point with coordinates (0, 0).
+- `Point2D(const Point2D& point)`: Copy constructor. Creates a new point by copying the coordinates of another point.
+- `Point2D(Point2D&& point)`: Move constructor. Creates a new point by moving the coordinates of another point.
+- `Point2D(value_type x, value_type y)`: Constructor. Initializes the point with the specified coordinates.
+- `Point2D(const OtherStructPoint2D auto& point)`: Initializes the point with the coordinates (`point.x`, `point.y`).
+- `Point2D(const ClassPoint2D auto& point)`: Initializes the point with the coordinates (`point.x()`, `point.y()`).
+- `template <class U, class V> Point2D(const std::pair<U, V>& pair)`: Constructor. Initializes the point with the
+  coordinates from a `std::pair`.
+
+### Static Constructors
+
+- `static Point2D origin()`: Returns a point representing the origin (0, 0).
+- `static Point2D fromPolar(value_type angle) requires(std::floating_point<value_type>)`: Creates a point from polar
+  coordinates with the given angle.
+- `static Point2D fromPolar(value_type angle, value_type length) requires(std::floating_point<value_type>)`: Creates a
+  point from polar coordinates with the given angle and length.
+
+### Validators
+
+- `bool isNull() const`: Checks if the point is null (coordinates are both zero). Returns `true`
```

Comment:
```suggestion
- `bool isOrigin() const`: Checks if the point is null (coordinates are both zero). Returns `true`
```
discard null word for this case.

Reply:
It was originally inspired on [`QPointF::isNull`](https://doc.qt.io/qt-6/qpointf.html#isNull), but I agree that isOrigin looks better for this purpose...

However, perhaps isNull is a common interface for other geometric representations such as [`QRectF`](https://doc.qt.io/qt-6/qrectf.html#isNull), [`QVector3D`](https://doc.qt.io/qt-6/qvector2d.html)...

wdyt?